Refactor parts of toric geometry

lkastner opened the pull request:
- `fan` renamed to `polyhedral_fan`
- Eliminate hack in `polyhedral_fan` that was used at some point since casting did not work
- The `polyhedral_fan` is also not stored as an attribute anymore, as this only duplicates information and the toric variety itself already provides the full fan functionality.
- Disable printing of `rays` of a fan in toric schemes part. We will never want that kind of detail in the printout except if someone calls rays directly. Also in case of a fan this information does not even describe the fan completely.
- Enable `--depwarn=error` for the doctests
- Fix doctests

lkastner committed:
ToricSchemes: Reduce printing

Does not adhere to
https://docs.oscar-system.org/stable/DeveloperDocumentation/printing_details/

lkastner committed:
Enable depwarn=error for doctests

lkastner committed:
ToricGeometry: Avoid another fan copy

lkastner committed:
ToricGeometry: Rework some constructors

HereAround reviewed (approved):
Thank you @lkastner  for those changes. In retrospect, I agree that we should not print the rays for the toric schemes. Sadly, this is the exact part that requires rebase/adjusting after I merged https://github.com/oscar-system/Oscar.jl/pull/2470. Sorry about that.

Other than that - thank you very much. Looks great!

HereAround commented:
I am actually surprised by `julia --project=docs --depwarn=error --color=yes -e'`. I recall making that change a while back. But maybe I only switched it on for the "real" tests and not the doctests? It is definitely good to have that.

lkastner committed:
ToricGeometry: fan->polyhedral_fan, fix deprecations

Also remove ugly hack that was in place before since casting via
Polymake.jl did not work. Now a true copy of the polyhedral fan is made
and returned.

Update all doctests, eliminate some other deprecations

lkastner committed:
FTheoryTools: Remove unnecessary calls of polyhedral_fan
